EFCore.NamingConventions/NamingConventions/Internal/SnakeCaseNameRewriter.cs

```diff
@@ -21,52 +22,52 @@ class SnakeCaseNameRewriter : IEntityTypeAddedConvention, IPropertyAddedConventi
             => propertyBuilder.HasColumnName(
                 ConvertToSnakeCase(propertyBuilder.Metadata.GetColumnName()));
 
-        static string ConvertToSnakeCase(string value)
+        static string ConvertToSnakeCase(string name)
         {
-            const char underscore = '_';
-            const UnicodeCategory noneCategory = UnicodeCategory.Control;
+            if (string.IsNullOrEmpty(name))
+                return name;
 
-            var builder = new StringBuilder();
-            var previousCategory = noneCategory;
+            var builder = new StringBuilder(name.Length + Math.Min(2, name.Length / 5));
+            var previousCategory = default(UnicodeCategory?);
 
-            for (var currentIndex = 0; currentIndex < value.Length; currentIndex++)
+            for (var currentIndex = 0; currentIndex < name.Length; currentIndex++)
             {
-                var currentChar = value[currentIndex];
-                if (currentChar == underscore)
+                var currentChar = name[currentIndex];
+                if (currentChar == '_')
                 {
-                    builder.Append(underscore);
-                    previousCategory = noneCategory;
+                    builder.Append('_');
+                    previousCategory = null;
                     continue;
                 }
 
                 var currentCategory = char.GetUnicodeCategory(currentChar);
                 switch (currentCategory)
                 {
-                case UnicodeCategory.UppercaseLetter:
-                case UnicodeCategory.TitlecaseLetter:
-                    if (previousCategory == UnicodeCategory.SpaceSeparator ||
-                        previousCategory == UnicodeCategory.LowercaseLetter ||
-                        previousCategory != UnicodeCategory.DecimalDigitNumber &&
-                        currentIndex > 0 &&
-                        currentIndex + 1 < value.Length &&
-                        char.IsLower(value[currentIndex + 1]))
-                    {
-                        builder.Append(underscore);
-                    }
+                    case UnicodeCategory.UppercaseLetter:
```

Comment:
I thought we liked unindented case labels in switches :)

Reply:
No, I don't like them, but following in Npgsql because it's the existing style (: Unindent them or leave them as is (use the common .NET style)?

As a side note, does it make sense to use `.editorconfig` from the main repo?

Reply:
Let's unindent :) I prefer it this way...

And yeah, .editorconfig would be good too.

Reply:
What option is responsible to this?